src/main/java/seedu/address/logic/parser/ParserType.java

```diff
@@ -1,19 +1,33 @@
 package seedu.address.logic.parser;
 
-import seedu.address.logic.commands.taskcommands.AddTaskCommand;
-import seedu.address.logic.commands.taskcommands.DeleteTaskCommand;
+import java.lang.reflect.Field;
 
-public abstract class ParserType {
+public class ParserType {
+		
+	private static final Class<?>[] parserTypes = new Class[]{AddCommandParser.class, DeleteCommandParser.class};
+	
 	public static CommandParser get(String commandWord){
-		 switch (commandWord) {
-	        case AddTaskCommand.COMMAND_WORD:
-	            return new AddCommandParser();
-
-	        case DeleteTaskCommand.COMMAND_WORD:
-	            return new DeleteCommandParser();
-
-	        default:
-	            return new IncorrectCommandParser();
-	        }
+		for(int i=0; i<parserTypes.length; i++){
+			try {
+				Field type = parserTypes[i].getField("COMMAND_WORD");
+				if(type.get(null).equals(commandWord)){
+					return (CommandParser)parserTypes[i].newInstance();
+				}
+			} 
+			catch (NoSuchFieldException e) {
+				return new IncorrectCommandParser();
+			} catch (SecurityException e) {
+				// TODO Auto-generated catch block
```

Comment:
I think perhaps we should either
1) Log this
or 
2) Force a program crash

because it's very unexpected if any of these happen from NoSuchField onwards. NoSuchField seems like a programmer error TBH - like we included an incorrect class in the list of parser classes - so just printing a stack trace may not do much